### PR TITLE
docs: add comprehensive docs for usage of grid | sf

### PR DESCRIPTION
**Pull request by StefanFeser**

### CODE REVIEW

File: src/components/Footer/Footer.module.scss [hunk 1]
```diff
@@ -19,6 +25,8 @@
 }
 
 .footer-content {
+  // Sets the footer-content element to be a main content area
+  // by using the predefined grid area 'main'.
```
Comment (SetupCoding) [inline src/components/Footer/Footer.module.scss]: For short and consistent wording (like before in page.module.scss), we could replace the comment with this one:
```suggestion
  // Places the footer-content element in the predefined "main" column.
```

File: src/components/Header/Header.module.scss [hunk 2]
```diff
@@ -1,28 +1,48 @@
...
 
+// **************************************************************
+// The STIHL Design System Grid mixin is used in this example app.
+// It is based on the native CSS Grid and provides a simple way to create
+// responsive grid layouts.
+// The STIHL Design System Grid is built to be used in a standard "row-based" layout
+// where every row is a grid container and every column is a grid item.
+// This allows developers to create complex, responsive layouts while maintaining
+// a clear and modular structure.
+// This example app demonstrates the usage of the STIHL Design System Grid.
```
Comment (SetupCoding) [inline src/components/Header/Header.module.scss]: I'd remove those two sentences. Feels "bloaty".

File: src/components/Header/Header.module.scss [hunk 3]
```diff
@@ -1,28 +1,48 @@
...
-  @include ds-grid-full-column;
+  // Initializes the grid layout for the top bar container.
+  // This enables us to place the contents inside the top bar along
+  // the predefined grid areas.
   @include ds-grid;
+
+  // DO WE NEED THIS? width is already set to 100vw ?
+  @include ds-grid-full-column;
+  // And this?! width is automatically set to 100% ?
   width: 100vw;
```
Comment (SetupCoding) [inline src/components/Header/Header.module.scss]: I'd remove width and keep the full-column, just to be "explicit" about what column the top-bar is located in. Makes it easier to identify its layout properties.

File: src/components/Header/Header.module.scss [hunk 4]
```diff
@@ -1,28 +1,48 @@
...
   min-height: max-content;
   background: $ds-color-core-grey-mid;
 }
 
 .top-bar-content {
+  // Sets the top-bar-content element to be a main content area
+  // by using the predefined grid area 'main'.
```
Comment (SetupCoding) [inline src/components/Header/Header.module.scss]:
```suggestion
  // Places the top-bar-content element in the predefined "main" column.
```

File: src/components/Header/Header.module.scss [hunk 5]
```diff
@@ -1,28 +1,48 @@
...
+  // Initializes the grid layout for the top bar container.
+  // This enables us to place the contents inside the hero section along
+  // the predefined grid areas.
   @include ds-grid;
-  width: 100vw;
 }
 
 .hero-image {
+  // To span the hero image section across the full width of the
+  // predefined "extended area", we use the ds-grid-extended-column mixin.
```
Comment (SetupCoding) [inline src/components/Header/Header.module.scss]:
```suggestion
  // predefined 'extended' area, we use the ds-grid-extended-column mixin.
```

File: src/components/Header/Header.module.scss [hunk 6]
```diff
@@ -70,6 +90,8 @@
 }
 
 .hero-image-title {
+  // Sets the white hero-image-title wrapper to be a main content area
+  // by using the predefined grid area 'main'.
```
Comment (SetupCoding) [inline src/components/Header/Header.module.scss]:
```suggestion
  // Places the hero-image-title-wrapper element in the predefined "main" column.
```

File: src/components/Header/Header.module.scss (hunk 3 above)
Comment (StefanFeser) [reply src/components/Header/Header.module.scss]: I understand. But it does not have any effect, right? For it to have an effect the parent element would need to be a ds-grid, too?

Comment (SetupCoding) [reply src/components/Header/Header.module.scss]: Yeah right, the parent has display:block on it. Then we can remove it :)

**Review by SetupCoding — APPROVED** [review]
📖

**Pull request merged**